facade/src/main/java/org/jboss/pnc/facade/providers/BuildProviderImpl.java

```diff
@@ -63,35 +77,56 @@
 
 @PermitAll
 @Stateless
+@Slf4j
 public class BuildProviderImpl extends AbstractProvider<BuildRecord, Build, BuildRef> implements BuildProvider {
 
     private BuildRecordRepository buildRecordRepository;
     private BuildConfigurationRepository buildConfigurationRepository;
     private BuildConfigurationAuditedRepository buildConfigurationAuditedRepository;
 
+    private BuildRecordPushResultRepository buildRecordPushResultRepository;
+
     private Gerrit gerrit;
     private BuildConfigurationRevisionMapper buildConfigurationRevisionMapper;
     private BuildMapper buildMapper;
 
     private BuildCoordinator buildCoordinator;
 
+    private BuildResultPushManager buildResultPushManager;
+    private BuildPushResultMapper buildPushResultMapper;
+
+    private AuthenticationProvider authenticationProvider;
+
+    @Context
+    private HttpServletRequest httpServletRequest;
```

Comment:
Can UserService be used instead of AuthenticationProvider & HttpServlerRequest?

Reply:
Done! Thanks for the suggestion! Didn't even know it was there :)

Thanks again!